middleware/route-guard.js

```diff
@@ -0,0 +1,23 @@
+//for the log IN part, to check if the user is already logged in:
+let isLoggedIn = (req,res,next) => {
+    //we check first if the user isn't logged in. if that's the case, then we redirect him to the loginpage instantly
+    if(!req.session.user) {
+        //if session is OFF, then redirected to the login page
+        res.redirect('/login');
+        next();
```

Comment:
What is causing your webpages that had the `isLoggedIn` or `isLoggedOut` methods to get stuck, is that you have the `next()` inside your `if` statement. You were checking if we had a 'req.session.user' or not, but then, you redirect the user and ALSO pass the `next()` inside your if statement, so, it would always leave the website stuck in the middleware.

```
let isLoggedIn = (req, res, next) => {
  //we check first if the user isn't logged in. if that's the case, then we redirect him to the loginpage instantly
  if (!req.session.user) {
    //if session is OFF, then redirected to the login page
    res.redirect('/login');
  }
   // only after we check our conditional, we pass the 'next()' method
  next();
};
```